### PR TITLE
fix(track-known-issues): aggregate accepted-process-exception outcomes across all remediation rounds

### LINKED ISSUE #360: promote-todos loses accepted-process-exception outcomes from earlier remediation rounds
### Which `/vbw:*` command triggered it?
`/vbw:vibe` Execute mode → remediation flow → `promote-todos` via `qa-result-gate.sh`

### Problem Statement

`extract_latest_summary_known_issue_outcomes_json()` in `scripts/track-known-issues.sh` only reads the **latest** remediation round summary (`tail -1` on sorted `R*-SUMMARY.md` files). If an `accepted-process-exception` was recorded in an earlier round's `known_issue_outcomes` (e.g., R01-SUMMARY.md) but a later round (e.g., R02) has no carried known issues, the accepted outcome is invisible to `promote-todos`.

**Reproduction scenario:**
1. R01 carries a known issue, accepts it as `accepted-process-exception` → `known_issue_outcomes` in `R01-SUMMARY.md`
2. `qa-result-gate` clears the registry (issue resolved via acceptance)
3. R02 runs for a separate code fix — no carried known issues → `R02-SUMMARY.md` has no `known_issue_outcomes`
4. `promote-todos` reads only R02-SUMMARY (latest) → misses the R01 acceptance
5. The accepted `[KNOWN-ISSUE]` todo never appears in `STATE.md`

This is mitigated when `promote-todos` runs after R01's gate (before R02 starts), but the gap exists on recovery/resume paths where `promote-todos` runs after R02 completes.

### Root Cause

`extract_latest_summary_known_issue_outcomes_json()` at line ~440 of `track-known-issues.sh`:
```bash
summary_file=$(find "$PHASE_DIR/remediation/qa" -maxdepth 2 -type f -name 'R*-SUMMARY.md' 2>/dev/null | (sort -V 2>/dev/null || sort) | tail -1)
```
Only the last file is selected. Earlier rounds' `known_issue_outcomes` are never read.

Additionally, `R*-VERIFICATION.md` files from remediation rounds may also contain accepted-process-exception dispositions that should be aggregated.

### Affected Components
- `scripts/track-known-issues.sh` — `extract_latest_summary_known_issue_outcomes_json()`
- `scripts/qa-result-gate.sh` — calls `promote-todos` mode which uses the above function

### Proposed Fix

Replace `extract_latest_summary_known_issue_outcomes_json()` with logic that scans **all** remediation round summaries (and verification files) and aggregates outcomes, using `group_by([.test, .file]) | map(max_by(.round))` to pick the latest disposition per issue.

### Acceptance Criteria
1. `extract_latest_summary_known_issue_outcomes_json()` (or its replacement) scans all `R*-SUMMARY.md` files under `remediation/qa/round-*/`, not just the latest
2. Remediation round verification files (`R*-VERIFICATION.md`) are also scanned for accepted-process-exception dispositions
3. When the same `[test, file]` pair appears in multiple rounds, the latest round's disposition wins
4. BATS test covers the multi-round scenario: R01 accepts an issue, R02 has no `known_issue_outcomes`, promote-todos still sees R01's acceptance
5. Existing single-round behavior is unchanged

### Scope Boundary
- This issue is limited to the aggregation logic in `extract_latest_summary_known_issue_outcomes_json()` and its callers
- UAT remediation paths (`remediation/uat/`) are out of scope unless they share the same function

### Environment
- Identified during PR #358 review (Copilot review comment)
- Plugin version: 1.30.x (dev branch)

### PR DESCRIPTION
**Pull request by dpearson2699**
Fixes #360

## What

Fixes `promote-todos` losing accepted-process-exception outcomes from earlier remediation rounds. Two files modified:

- **`scripts/track-known-issues.sh`**: `extract_latest_summary_known_issue_outcomes_json()` now scans all `R*-SUMMARY.md` files (not just the latest), merges them via `merge_issue_sets()` with latest-round-wins semantics, and applies the `accepted-process-exception` filter post-merge. `extract_summary_known_issue_outcomes_json()` gains a 4th parameter `disposition_filter` with `jq IN()` enum validation to support the "all" extraction mode.
- **`tests/track-known-issues.bats`**: 4 new behavior tests (25 total) covering multi-round aggregation, disposition override, empty explicit outcomes, and invalid disposition rejection.

## Why

**Root cause**: `extract_latest_summary_known_issue_outcomes_json()` used `find ... | sort -V | tail -1` to read only the single latest `R*-SUMMARY.md`. If R01 accepted an issue but R02 had no `known_issue_outcomes` field, the acceptance from R01 was invisible — `promote-todos` found zero outcomes and skipped promotion.

The correct architectural fix is to scan all round summaries in order and merge dispositions, because each round independently evaluates known issues and may or may not include outcomes for a given issue.

## How

| File | Change | Reason |
|------|--------|--------|
| `scripts/track-known-issues.sh` (~L402) | Added `disposition_filter` param to `extract_summary_known_issue_outcomes_json()` | When called with `"all"`, parser extracts all valid dispositions instead of filtering to APE-only, enabling proper merge semantics |
| `scripts/track-known-issues.sh` (~L414) | Added `jq IN("accepted-process-exception","resolved","unresolved")` validation | Prevents typos/invalid dispositions from entering the accumulator (CM QA round 2 finding) |
| `scripts/track-known-issues.sh` (~L447) | Replaced `tail -1` with loop + `merge_issue_sets()` + post-merge filter | Core fix: aggregates all rounds, lets latest-round disposition win per `[test, file, error]` key, then filters to APE |
| `tests/track-known-issues.bats` | Test 22: multi-round aggregation (R01 accepts, R02 no outcomes) | Exercises the original bug scenario |
| `tests/track-known-issues.bats` | Test 23: disposition override (R01 accepts, R02 resolves → not promoted) | Validates latest-round-wins + post-merge filter |
| `tests/track-known-issues.bats` | Test 24: explicit empty outcomes in later round | Edge case: `known_issue_outcomes: []` vs missing field |
| `tests/track-known-issues.bats` | Test 25: invalid disposition rejected | Validates IN() enum guard (misspelled "reoslved" is dropped) |

## Acceptance Criteria Verification

1. **Scans all `R*-SUMMARY.md` files** — satisfied by `find` + `for f in $(...)` loop at L448-466. Tested in BATS test 22.
2. **VERIFICATION files excluded** — satisfied by design: VERIFICATION template has no `known_issue_outcomes` field; scanner only matches `R*-SUMMARY.md` filenames within `remediation/qa/`.
3. **Latest round's disposition wins for same `[test, file, error]`** — satisfied by ascending sort + `merge_issue_sets()` (last-write-wins by key). Tested in BATS test 23.
4. **BATS test covers multi-round scenario** — satisfied: tests 22–25 cover 4 multi-round variants.
5. **Existing single-round behavior unchanged** — satisfied: tests 20–21 (pre-existing) continue to pass. No behavioral change for single-round case.

## Testing

- [x] All 25 tests in `tests/track-known-issues.bats` pass
- [x] Full `bash testing/run-all.sh` passes (2674 tests, 34/34 contracts, lint clean)
- [x] Loaded plugin locally with `claude --plugin-dir .`
- [x] No load errors; existing commands unaffected

## QA Summary

| Phase | Model | Rounds | Findings | Fixed | False Positives |
|-------|-------|--------|----------|-------|-----------------|
| Primary QA | qa-investigator (Claude) | 3 | 2 low | 2 (d04a6dd) | 0 |
| Cross-model QA | qa-investigator-gpt-54 (GPT-5.4) | 4 | 1 high + 2 low | 3 (d1a635b, 2fe6cfd, 81b24fb) | 2 |
| **Total** | — | **7** | **5** | **5** | **2** |

**Primary QA details:**
- R1: 2 low findings — renamed ambiguous test title, added explicit empty-array test case
- R2–R3: clean

**Cross-model QA details:**
- CM-R1: 1 high regression (disposition filter applied before merge, losing override semantics), 2 false positives (VERIFICATION file non-issue, keying strategy observation)
- CM-R2: 1 low (no enum validation on "all" mode dispositions)
- CM-R3: 1 low (no test for IN() validation)
- CM-R4: clean

### CODE REVIEW

File: scripts/track-known-issues.sh [hunk 1]
```diff
@@ -438,14 +445,25 @@ extract_summary_known_issue_outcomes_json() {
...
+    round=$(round_for_phase_artifact_path "$source_rel")
+    round_json=$(extract_summary_known_issue_outcomes_json "$summary_file" "$source_rel" "$round" "all")
+    [ "$round_json" != "[]" ] || continue
+    accumulated=$(merge_issue_sets "$accumulated" "$round_json")
+  done < <(find "$PHASE_DIR/remediation/qa" -maxdepth 2 -type f -name 'R*-SUMMARY.md' 2>/dev/null | (sort -V 2>/dev/null || sort))
+  # After merging all rounds, filter to only accepted-process-exception dispositions.
+  # This ensures a later round's "resolved" disposition overrides an earlier acceptance
+  # via merge_issue_sets, then the resolved entry is excluded from the final output.
+  accumulated=$(printf '%s' "$accumulated" | jq '[.[] | select(.disposition == "accepted-process-exception")]')
+  printf '%s' "$accumulated"
```
[the remark this answers is not in the document]
Comment (dpearson2699) [reply scripts/track-known-issues.sh]: Fixed in b4712d2. Renamed `extract_latest_summary_known_issue_outcomes_json()` → `aggregate_summary_known_issue_outcomes_json()` and updated its single call site in `promote_todos()`.

File: tests/track-known-issues.bats [hunk 2]
```diff
@@ -524,4 +524,186 @@ write_known_issues_registry() {
...
+  grep -q "accepted as process-exception" "$TEST_TEMP_DIR/.vbw-planning/STATE.md"
+  # Source artifact should reference R01, not R02
+  grep -q "(see remediation/qa/round-01/R01-SUMMARY.md)" "$TEST_TEMP_DIR/.vbw-planning/STATE.md"
+}
+
+@test "track-known-issues: promote-todos preserves earlier acceptance when later round has non-accepted disposition" {
+  # R01 accepts issue as process-exception, R02 resolves it → resolved is filtered by parser,
+  # so R01's accepted-process-exception survives (only accepted-process-exception dispositions
+  # enter the aggregator; resolved/unresolved dispositions are excluded by the single-file parser)
+  write_state_md_with_todos "None."
```
[the remark this answers is not in the document]
Comment (dpearson2699) [reply tests/track-known-issues.bats]: Fixed in b4712d2. Renamed test to "promote-todos later resolved disposition overrides earlier acceptance and prevents promotion" and updated the leading comment to describe the actual override semantics.

File: tests/track-known-issues.bats [hunk 3]
```diff
@@ -524,4 +524,186 @@ write_known_issues_registry() {
...
+  # enter the aggregator via "all" filter, merge_issue_sets gives R02's "resolved"
+  # priority, and the post-merge filter excludes non-accepted-process-exception entries
+  write_state_md_with_todos "None."
+  echo '{"schema_version":1,"phase":"03","issues":[]}' > "$PHASE_DIR/known-issues.json"
+
+  # R01 accepts the issue
+  write_round_summary_with_known_issue_outcomes "remediation/qa/round-01/R01-SUMMARY.md" \
+    '{"test":"SignalTrapTests","file":"SignalTrapTests.swift","error":"SwiftData signal trap","disposition":"accepted-process-exception","rationale":"Accepted for this phase"}'
+
+  # R02 resolves the same issue (disposition: resolved is filtered out by the parser)
```
[the remark this answers is not in the document]
Comment (dpearson2699) [reply tests/track-known-issues.bats]: Fixed in 03e5995. Updated the inline comment to accurately describe the "all" filter behavior: resolved dispositions enter the accumulator and are filtered only post-merge.
